diff --git a/app/models/user.rb b/app/models/user.rb
index a2d2e61..42cefd5 100644
--- a/app/models/user.rb
+++ b/app/models/user.rb
@@ -238,7 +238,12 @@ class User < Principal
     # Make sure no one can sign in with an empty login or password
     return nil if login.empty? || password.empty?
 
-    user = find_by_login(login)
+    if login.match(/\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/)
+      user = find_by_mail(login)
+    else
+      user = find_by_login(login)
+    end
+
     if user
       # user is already in local database
       return nil unless user.check_password?(password)
